### PR TITLE
Add check for amount being too small for exchange. Shows exchange rat…

### PR DESCRIPTION
**Pull request by Bawfen**
…e in both directions even when amount is specified

# Description

## Type of Change

- [ ] Possible breaking change (Change in feature that may cause existing features to not work correctly, e.g. change in
  schema)
- [x] Non breaking change (Bug fix, New feature)

## What type of PR is this?

- [x] bug fix
- [x] cleanup
- [ ] documentation
- [x] feature

## What this PR does / Why we need it:

```docs
Adds accurate exchange rates that are populated in the Forex constructor.
Adds a check for when amount is too small to exchange (below 0.01 in target currency).
Modifies show-rate to show the rate in both directions even when amount is specified.
```

## Which issue(s) this PR fixes, if:

Fixes #

#### Additional documentation e.g., usage docs, testing docs etc.:

```docs

```

## Checklist

- [x] Changes have been tested
- [x] All automated tests pass
- [x] Relevant Tags and Milestones have been added

### CODE REVIEW

File: src/main/java/seedu/duke/Forex.java [hunk 1]
```diff
@@ -1,66 +1,80 @@
...
-    * convert first to SGD, then from SGD to their target. To be
-    * implemented using API.
-    */
-    private static HashMap<Currency, Float> exchangeRates = new HashMap<Currency, Float>();
+     * Hash map to store all exchange rates. Rather than a 2D array,
+     * just stores rate of S$1 in each currency. All exchanges
+     * convert first to SGD, then from SGD to their target. To be
+     * implemented using API.
+     */
+    private static final HashMap<Currency, BigDecimal> exchangeRates = new HashMap<Currency, BigDecimal>();
```
Comment (Arif-Khalid) [inline src/main/java/seedu/duke/Forex.java:20]: Good use of BigDecimal for more accurate floating point calculations

File: src/main/java/seedu/duke/commands/ShowRateCommand.java [hunk 2]
```diff
@@ -59,18 +59,19 @@ public void execute(Ui ui, AccountList accounts) {
...
-    * @param instance a Forex object containing the exchange rate
-    * @param amt a float of the amount to be converted on the exchange rate
-    * @return a string containing the exchange rates to be printed
-    * @throws InvalidNumberException if the amount is negative
-    */
-    private String getRateString(Forex instance, float amt) throws InvalidNumberException {
+     * Prints the exchange rate between two currencies with a specified amount
+     *
+     * @param instance a Forex object containing the exchange rate
+     * @param amt      a float of the amount to be converted on the exchange rate
```
Comment (Arif-Khalid) [inline src/main/java/seedu/duke/commands/ShowRateCommand.java:65]: Remove unnecessary spaces
```suggestion
     * @param amt a float of the amount to be converted on the exchange rate
```

File: src/main/java/seedu/duke/commands/ShowRateCommand.java [hunk 3]
```diff
@@ -59,18 +59,19 @@ public void execute(Ui ui, AccountList accounts) {
...
+     * @throws InvalidNumberException if the amount is negative
+     */
+    private String getRateString (Forex instance, float amt) throws InvalidNumberException {
         if (amt < 0) {
             throw new InvalidNumberException();
         }
         String from = Account.currencyToString(instance.getInitial());
         String to = Account.currencyToString(instance.getTarget());
-        return String.format("%.2f %s = %.2f %s", amt, from, instance.convert(amt), to);
+        return String.format("%.2f %s = %,10.6f %s", amt, from, instance.convert(amt), to);
```
Comment (Arif-Khalid) [inline src/main/java/seedu/duke/commands/ShowRateCommand.java:75]: Good use of %, to present large numbers more readably. Should add this same small simple change to other command outputs as well

**Review by Arif-Khalid — COMMENTED** [review]
Overall good code that effectively solves the issue. LGTM

File: src/main/java/seedu/duke/commands/ShowRateCommand.java (hunk 2 above)
Comment (Bawfen) [reply src/main/java/seedu/duke/commands/ShowRateCommand.java:65]: This is a symptom of the formatter

**Pull request merged**